### PR TITLE
Add a `restore` command

### PR DESCRIPTION
**Pull request by sharder996**

### CODE REVIEW

**Review by ricab — COMMENTED** [review]
Nice job. A few minor details inline.

**Comment by ricab** [comment]
Also, this does not cover the messages for the auto snapshot:

```
Taking snapshot: <instance>.snapshotX /
Snapshot taken: <instance>.<snapshot>
```

We can do that in a separate PR though. I will add an item to the board.

File: src/client/cli/cmd/restore.cpp [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+    if (num_args > 1)
+    {
+        cerr << "Too many arguments supplied\n";
+        return ParseCode::CommandLineError;
+    }
+
+    auto instance = parser->positionalArguments().at(0);
+    auto instance_name = instance.section('.', 0, 0);
+    auto snapshot_name = instance.section('.', 1);
```
[the remark this answers is not in the document]
Comment (sharder996) [reply src/client/cli/cmd/restore.cpp]: `splitRef` isn't in the docs for Qt6 anymore so I used `split` instead.

File: src/client/cli/cmd/restore.cpp [hunk 2]
```diff
@@ -97,28 +97,20 @@ mp::ParseCode cmd::Restore::parse_args(mp::ArgParser* parser)
...
-        cerr << "Need the name of an instance to restore.\n";
-        return ParseCode::CommandLineError;
-    }
-
-    if (snapshot_name.isEmpty())
+    const auto tokens = parser->positionalArguments().at(0).split('.');
+    if (tokens.size() != 2 || tokens[0].isEmpty() || tokens[1].isEmpty())
     {
-        cerr << "Need the name of a snapshot to restore from.\n";
+        cerr << "Invalid format. Specify the instance to restore and snapshot to use in the form "
```
Comment (ricab) [inline src/client/cli/cmd/restore.cpp:107]:
```suggestion
        cerr << "Invalid format. Please specify the instance to restore and snapshot to use in the form "
```

**Review by ricab — COMMENTED** [review]
Cool, thanks.

**Pull request merged**